conans/client/conf/config_installer.py

```diff
@@ -111,32 +111,65 @@ def _process_download(item, client_cache, output, tmp_folder, verify_ssl, reques
         raise ConanException("Error while installing config from %s\n%s" % (item, str(e)))
 
 
-def configuration_install(item, client_cache, output, verify_ssl, requester,
+def configuration_install(path_or_url, client_cache, output, verify_ssl, requester,
                           config_type=None, args=None):
     tmp_folder = os.path.join(client_cache.conan_folder, "tmp_config_install")
     # necessary for Mac OSX, where the temp folders in /var/ are symlinks to /private/var/
     tmp_folder = os.path.realpath(tmp_folder)
     mkdir(tmp_folder)
     try:
-        if item is None:
+        if path_or_url is None:
             try:
                 item = client_cache.conan_config.get_item("general.config_install")
+                _config_type, path_or_url, _args = _process_config_install_item(item)
             except ConanException:
                 raise ConanException("Called config install without arguments and "
                                      "'general.config_install' not defined in conan.conf")
-
-        if item.endswith(".git") or config_type == "git":
-            _process_git_repo(item, client_cache, output, tmp_folder, verify_ssl, args)
-        elif os.path.isdir(item):
-            _process_folder(item, client_cache, output)
-        elif os.path.isfile(item):
-            _process_zip_file(item, client_cache, output, tmp_folder)
-        elif item.startswith("http"):
-            _process_download(item, client_cache, output, tmp_folder, verify_ssl,
+        else:
+            _config_type, path_or_url, _args = _process_config_install_item(path_or_url)
+
+        config_type = config_type or _config_type
+        args = args or _args
+
+        if os.path.exists(path_or_url):
+            path_or_url = os.path.abspath(path_or_url)
+
+        if config_type == "git":
+            _process_git_repo(path_or_url, client_cache, output, tmp_folder, verify_ssl, args)
+        elif config_type == "dir":
+            args = None
+            _process_folder(path_or_url, client_cache, output)
+        elif config_type == "file":
+            args = None
+            _process_zip_file(path_or_url, client_cache, output, tmp_folder)
+        elif config_type == "url":
+            args = None
+            _process_download(path_or_url, client_cache, output, tmp_folder, verify_ssl,
                               requester=requester)
         else:
-            raise ConanException("I don't know how to process %s" % item)
+            raise ConanException("Unable to process config install: %s" % path_or_url)
     finally:
-        if item:
-            client_cache.conan_config.set_item("general.config_install", item)
+        value = "%s:[%s, %s]" % (config_type, path_or_url, args)
+        client_cache.conan_config.set_item("general.config_install", value)
         rmdir(tmp_folder)
+
+
+def _process_config_install_item(item):
```

Comment:
Would be nice to have a docstring with some input examples. At first sigh I don't know what this line could be doing: ` [item.strip() for item in rest[1:-1].rstrip().split(",", 1)]`